fetcher/fedora.go

```diff
@@ -0,0 +1,293 @@
+		return nil, err
+	}
+
+	updates, err := fetchUpdateInfosFedora(feeds)
+	if err != nil {
+		return nil, err
+	}
+
+	moduleYaml, err := fetchModulesYamlFedora(feeds)
+	if err != nil {
+		return nil, err
+	}
+
+	results, err := parseFetchResultsFedora(updates)
+	if err != nil {
+		return nil, err
+	}
+
+	for version, result := range results {
+		log15.Info(fmt.Sprintf("%d CVEs for Fedora %s modules Fetched", len(result.UpdateList), version))
+		for i, update := range result.UpdateList {
+			yml, ok := moduleYaml[version][update.Title]
+			if ok {
+				var pkgs []Package
+				for _, rpm := range yml.Data.Artifacts.Rpms {
+					pkg, err := rpm.NewPackageFromRpm()
+					if err != nil {
+						return nil, err
+					}
+					pkgs = append(pkgs, pkg)
+				}
+				results[version].UpdateList[i].Packages = pkgs
+				results[version].UpdateList[i].ModularityLabel = yml.ConvertToModularityLabel()
+			}
+		}
+	}
+	return results, nil
+}
+
+func fetchFeedFilesFedora(reqs []fetchRequest) ([]FetchResult, error) {
+	if len(reqs) == 0 {
+		return nil, fmt.Errorf("There are no versions to fetch")
+	}
+	results, err := fetchFeedFiles(reqs)
+	if err != nil {
+		return nil, xerrors.Errorf("Failed to fetch. err: %w", err)
+	}
+	return results, nil
+}
+
+func fetchUpdateInfosFedora(results []FetchResult) ([]FetchResult, error) {
+	log15.Info("start fetch updateinfo in repomd.xml")
+	var updateInfoReqs []fetchRequest
+	for _, r := range results {
+		var repoMd RepoMd
+		if err := xml.NewDecoder(bytes.NewBuffer(r.Body)).Decode(&repoMd); err != nil {
+			log15.Warn(fmt.Sprintf("Failed to decode repomd. Skip to fetch version %s", r.Target), "err", err)
+			continue
+		}
+
+		for _, repo := range repoMd.RepoList {
+			if repo.Type == "updateinfo" {
+				u, err := url.Parse(r.URL)
+				if err != nil {
+					return nil, err
+				}
+				u.Path = strings.Replace(u.Path, "repodata/repomd.xml", repo.Location.Href, 1)
+				req := fetchRequest{
+					url:          u.String(),
+					target:       r.Target,
+					mimeType:     mimeTypeXz,
+					concurrently: false,
+				}
+				updateInfoReqs = append(updateInfoReqs, req)
+				break
+			}
+		}
+	}
+
+	if len(updateInfoReqs) == 0 {
+		return nil, fmt.Errorf("No updateinfo field in the repomd")
+	}
+
+	results, err := fetchFeedFiles(updateInfoReqs)
+	if err != nil {
+		return nil, xerrors.Errorf("Failed to fetch. err: %w", err)
+	}
+	return results, nil
+}
+
+func parseFetchResultsFedora(results []FetchResult) (fedoraUpdatesPerVersion, error) {
+	updateInfos := make(fedoraUpdatesPerVersion, len(results))
+	for _, r := range results {
+		var updateInfo FedoraUpdates
+		if err := xml.NewDecoder(bytes.NewReader(r.Body)).Decode(&updateInfo); err != nil {
+			return nil, err
+		}
+		var securityUpdate []FedoraUpdateInfo
+		for _, update := range updateInfo.UpdateList {
+			if update.Type == "security" {
+				cveIDs := []string{}
+				for _, ref := range update.References {
+					id := util.CveIDPattern.FindAllString(ref.Title, -1)
```

Comment:
The following CVE-IDs are listed in ref.Tile.
- CVE-2021-2478 
- CVE-2021-2479 
- CVE-2021-2481 
- CVE-2021-35546 
- CVE-2021-35575 
- CVE-2021-35577 
- CVE-2021-35591 
- CVE-2021-35596 
- CVE-2021-35597 
- CVE-2021-35602 
- CVE-2021-35604 
- CVE-2021-35607 
- CVE-2021-35608 
- CVE-2021-35610

```xml
<update from="updates@fedoraproject.org" status="stable" type="security" version="2.0">
    <id>FEDORA-2021-5ffabdc080</id>
    <title>chromium-94.0.4606.81-1.fc35</title>
    <references>
        <reference href="https://bugzilla.redhat.com/show_bug.cgi?id=2012664" id="2012664" type="bugzilla" title="CVE-2021-37977 CVE-2021-37978 CVE-2021-37979 CVE-2021-37980 chromium: various flaws [fedora-all]"/>
    </references>
<update>

<update from="updates@fedoraproject.org" status="stable" type="security" version="2.0">
    <id>FEDORA-2021-46dc82116b</id>
    <title>community-mysql-8.0.27-1.fc35</title>
    <references>
        <reference href="https://bugzilla.redhat.com/show_bug.cgi?id=2016141" id="2016141" type="bugzilla" title="CVE-2021-2478 CVE-2021-2479 CVE-2021-2481 CVE-2021-35546 CVE-2021-35575 CVE-2021-35577 CVE-2021-35591 CVE-2021-35596 CVE-2021-35597 CVE-2021-35602 CVE-2021-35604 CVE-2021-35607 CVE-2021-35608 CVE-2021-35610 ... community-mysql: various flaws [fedora-all]"/>
    </references>
<update
```
However, when I actually look at the Bugzilla link, there are more CVE-IDs tied to it than are shown in the ref.Title. (In some cases, CVE-IDs are omitted in ref.)
![Screenshot from 2022-01-19 14-07-24](https://user-images.githubusercontent.com/16035056/150067760-ac716f82-90d6-4b7a-b4bd-9f34aa9f2067.png)

If the CVE-ID is omitted, you need to traverse Bugzilla and collect all relevant CVE-IDs.